src/transformers/utils/__init__.py

```diff
@@ -274,3 +276,31 @@ def check_min_version(min_version):
             + "Check out https://github.com/huggingface/transformers/tree/main/examples#important-note for the examples corresponding to other "
             "versions of HuggingFace Transformers."
         )
+
+
+@lru_cache()
+def get_available_devices():
```

Comment:
Will this be considered a part of the public `transformers.utils` API in the future?

Reply:
In that case:

* should it be in the `_import_structure` shenanigans in `transformers/__init__.py`?
* it should have a type annotation
* it should return an immutable `frozenset`